File: simple_xor_with_c/xor.c
#include<stdio.h>
#include <string.h>
#include <ctype.h>
void encode_(char *word,int key,char *cipher){
    int len = strlen(word);
    for (int i= 0;i<len;i++){
        cipher[i]=(word[i]^key);
        //printf("%s\n",cipher[i]);
    }
    cipher[len]='\0';
    
}

void decode_(char *cipher,int key,char *decoded_string){
    int len = strlen(cipher);
    for(int i= 0;i < len ; i++){
        decoded_string[i]= cipher[i]^key;
    }
    decoded_string[len]='\0';
    printf("%d",len);
     
}
int main(){
    
    char word[] = "hello";
    int key = 2456;
    char cipher[strlen(word)+1];
    char decoded_string[strlen(word)+1];
    encode_(word,key,cipher);
    printf("%s\n",cipher);
    decode_(cipher,key,decoded_string);
    printf("%s\n",decoded_string);
    
    return 0;
}

File: simple_xor_with_c/xor.py
#string = 'K3y5_4r3_1mport4nT'
#key = 9823



def encode(string,key):
    cipher = ''
    for ch in string:
        cipher+=chr(ord(ch)^key)
    return cipher

def decode(cipher,key):
    dec = ''
    for ch in cipher:
        dec+=chr(ord(ch)^key)
    return dec

#cipher = encode(string,key)
cipher = '☔♬☦♪☀♫☭♬☀♮☲☯☰☭☫♫☱☋'

print(f"Try decoding this : {cipher}")
key = int(input("Hey this can give me the key to unlock the password: "))

dec = decode(cipher,key)

#print(cipher)
print(dec)


